### PR TITLE
📖 docs/../developers: describe metrics gathering for e2e tests

### PR DESCRIPTION
**Pull request by s-urbaniak**
## Summary
This describes how to inspect and gather Prometheus metrics for e2e test runs.

## Related issue(s)

Fixes https://github.com/kcp-dev/kcp/issues/2706

### CODE REVIEW

File: docs/content/en/developers/metrics.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+linkTitle: "kcp Prometheus metrics"
```
Comment (ncdc) [inline docs/content/en/developers/metrics.md:2]: We aren't using `linkTitle` any more

File: docs/content/en/developers/metrics.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+weight: 1
```
Comment (ncdc) [inline docs/content/en/developers/metrics.md:3]: We aren't using `weight` any more

File: docs/content/en/developers/metrics.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+description: >
+Inspecting Prometheus metrics of e2e runs
```
Comment (ncdc) [inline docs/content/en/developers/metrics.md:5]:
```suggestion
    Inspecting Prometheus metrics of e2e runs
```

File: docs/content/en/developers/metrics.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+# Inspecting Prometheus metrics for e2e runs
```
Comment (ncdc) [inline docs/content/en/developers/metrics.md:8]: Please name the file to roughly match with this heading. mkdocs sorts based on filename. So `inspecting-metrics-for-e2e-tests` or something like that.

File: docs/content/en/developers/metrics.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+For e2e tests, the above environment variable is respected by the following test servers:
+
+- when a shared kcp server is spawned using `SharedKcpServer` or `PrivateKcpServer` in the `kcp.test.e2e.framework` package.
+- `cmd/sharded-test-server`
+- `cmd/test-server/kcp`
+
+## Inspecting metrics of e2e runs
+
+### Github
```
Comment (ncdc) [inline docs/content/en/developers/metrics.md:25]: global: GitHub

**Review by ncdc — CHANGES_REQUESTED** [review]

File: docs/content/en/developers/metrics.md (hunk 1 above)
Comment (s-urbaniak) [reply docs/content/en/developers/metrics.md:2]: removed

File: docs/content/en/developers/metrics.md (hunk 2 above)
Comment (s-urbaniak) [reply docs/content/en/developers/metrics.md:3]: removed

File: docs/content/en/developers/metrics.md (hunk 3 above)
Comment (s-urbaniak) [reply docs/content/en/developers/metrics.md:5]: indented

File: docs/content/en/developers/metrics.md (hunk 5 above)
Comment (s-urbaniak) [reply docs/content/en/developers/metrics.md:25]: fixed

**Pull request merged**